sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TablesAsyncClientTest.java

```diff
@@ -0,0 +1,337 @@
+            })
+            .expectComplete()
+            .verify();
+    }
+
+    @Test
+    void deleteEntityWithResponseMatchEtag() {
+        // Arrange
+        String partitionKeyValue = testResourceNamer.randomName("partitionKey", 20);
+        String rowKeyValue = testResourceNamer.randomName("rowKey", 20);
+        Entity tableEntity = new Entity(partitionKeyValue, rowKeyValue);
+        tableEntity = asyncClient.createEntity(tableEntity).block();
+        int expectedStatusCode = 204;
+
+        // Act & Assert
+        StepVerifier.create(asyncClient.deleteEntityWithResponse(tableEntity, true))
+            .assertNext(response -> {
+                Assertions.assertEquals(expectedStatusCode, response.getStatusCode());
+            })
+            .expectComplete()
+            .verify();
+    }
+
+    @Test
+    void getEntityWithResponseTHIS() {
+        // Arrange
+        String partitionKeyValue = testResourceNamer.randomName("partitionKey", 20);
+        String rowKeyValue = testResourceNamer.randomName("rowKey", 20);
+        Entity tableEntity = new Entity(partitionKeyValue, rowKeyValue);
+        tableEntity = asyncClient.createEntity(tableEntity).block();
+        int expectedStatusCode = 200;
+        System.out.println("HERE " + tableEntity.getETag());
+
+        // Act & Assert
+        StepVerifier.create(asyncClient.getEntityWithResponse(tableEntity.getPartitionKey().toString(), tableEntity.getRowKey().toString()))
+            .assertNext(response -> {
+                Assertions.assertEquals(expectedStatusCode, response.getStatusCode());
+                System.out.println("ODATA " + response.getHeaders().get("etag"));
+            })
+            .expectComplete()
+            .verify();
+    }
+
+    /**
+     * expect to see both propertyA and propertyB since the UpdateMode is MERGE
+     */
+    @Test
+    void updateEntityWithResponseMerge() {
+        // Arrange
+        String partitionKeyValue = testResourceNamer.randomName("ApartitionKey", 20);
+        String rowKeyValue = testResourceNamer.randomName("ArowKey", 20);
+        Entity tableEntity = new Entity(partitionKeyValue, rowKeyValue);
+        tableEntity.addProperties("propertyA", "valueA");
+        tableEntity = asyncClient.createEntity(tableEntity).block();
+        tableEntity.getProperties().remove("propertyA");
+        tableEntity.addProperties("propertyB", "valueB");
+        int expectedStatusCode = 204;
+
+        // Act & Assert
+        StepVerifier.create(asyncClient.updateEntityWithResponse(tableEntity, true, UpdateMode.MERGE))
+            .assertNext(response -> {
+                Assertions.assertEquals(expectedStatusCode, response.getStatusCode());
+            })
+            .expectComplete()
+            .verify();
+    }
+
+    /**
+     * expect to see both propertyA and propertyB since the UpdateMode is MERGE
+     */
+    @Test
+    void updateEntityWithResponseMergeUnconditional() {
```

Comment:
@conniey This test is failing because the get isn't working, I could delete these tests until we get it fixed?